### DISCUSSION TITLE
About sinks global config

### DISCUSSION THREAD

**Discussion by yangshike** [vectordotdev/vector#17959] [General]
I need to write many clickhouse tables, but there is only one instance of clickhouse and database. However, now in the configuration file, I have to write the database address, username, and password for each table. Can there be a global configuration?

**Comment by jszwedko** [comment]
Hi @yangshike ,

There isn't a way to configure that globally in Vector but you could consider using a templating language to generate the Vector configuration that would allow more re-use. https://jsonnet.org/ is an example of one such language.

**Comment by yangshike** [comment]
😅
